integration_tests/ctesque/src/sharedTest/java/android/graphics/BitmapTest.java

```diff
@@ -359,6 +362,209 @@ public void createBitmapWithOffsetAndStride() {
+                .isMutable())
+        .isTrue();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      assertThat(
+              Bitmap.createBitmap(
+                      (DisplayMetrics) null,
+                      /* width= */ 1,
+                      /* height= */ 1,
+                      Bitmap.Config.ARGB_8888)
+                  .isMutable())
+          .isTrue();
+    }
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      assertThat(
+              Bitmap.createBitmap(
+                      /* width= */ 1,
+                      /* height= */ 1,
+                      Bitmap.Config.ARGB_8888,
+                      /* hasAlpha= */ true)
+                  .isMutable())
+          .isTrue();
+      assertThat(
+              Bitmap.createBitmap(
+                      /* displayMetrics= */ null,
+                      /* width= */ 1,
+                      /* height= */ 1,
+                      Bitmap.Config.ARGB_8888,
+                      /* hasAlpha= */ true)
+                  .isMutable())
+          .isTrue();
+    }
+
+    // Immutable constructor variants.
+    assertThat(
+            Bitmap.createBitmap(
+                    /* colors= */ new int[] {0},
+                    /* width= */ 1,
+                    /* height= */ 1,
+                    Bitmap.Config.ARGB_8888)
+                .isMutable())
+        .isFalse();
+    assertThat(
+            Bitmap.createBitmap(
+                    /* colors= */ new int[] {0},
+                    /* offset= */ 0,
+                    /* stride= */ 1,
+                    /* width= */ 1,
+                    /* height= */ 1,
+                    Bitmap.Config.ARGB_8888)
+                .isMutable())
+        .isFalse();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      assertThat(
+              Bitmap.createBitmap(
+                      /* displayMetrics= */ null,
+                      /* colors= */ new int[] {0},
+                      /* width= */ 1,
+                      /* height= */ 1,
+                      Bitmap.Config.ARGB_8888)
+                  .isMutable())
+          .isFalse();
+      assertThat(
+              Bitmap.createBitmap(
+                      /* displayMetrics= */ null,
+                      /* colors= */ new int[] {0},
+                      /* offset= */ 0,
+                      /* stride= */ 1,
+                      /* width= */ 1,
+                      /* height= */ 1,
+                      Bitmap.Config.ARGB_8888)
+                  .isMutable())
+          .isFalse();
+    }
+  }
+
+  @Test
+  public void createBitmap_hasAlpha() {
+    // ARGB_8888 has alpha by default.
+    assertThat(
+            Bitmap.createBitmap(/* width= */ 1, /* height= */ 1, Bitmap.Config.ARGB_8888)
+                .hasAlpha())
+        .isTrue();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
IMO it is not recommended to add multiple "if statement" inner test to test multiple occasions. If we move those "if statements` outer and split this test to multiple tests, it maybe brings redundant code for common assertions. Oh, version related assertions only test `Bitmap.createBitmap`, maybe we can move them outer to single test.